### PR TITLE
fix: allow parsing multiple requests from the same http connection

### PR DESCRIPTION
**Pull request by romange**

### CODE REVIEW

File: util/http/http_handler.cc [hunk 1]
```diff
@@ -315,7 +315,7 @@ error_code HttpConnection::ParseFromBuffer(io::Bytes buf) {
   cntx.set_user_data(user_data_);
 
   while (!buf.empty()) {
-    ParserType parser{std::move(request)};
+    ParserType parser;
```
Comment (chakaz) [inline util/http/http_handler.cc:318]: Is `request` not used by the `parser`?

Comment (romange) [reply util/http/http_handler.cc:318]: parser returns an parsed requests few lines below. without this change, the parser has reused already parsed requests for parsing it again, so it's good I added multiple requests to the tests - they uncovered the bug.

Comment (romange) [reply util/http/http_handler.cc:318]: see line 325

Comment (chakaz) [reply util/http/http_handler.cc:318]: gotcha, nice!

**Review by chakaz — APPROVED** [review]

**Pull request merged**